src/drf_yasg/renderers.py

```diff
@@ -64,10 +65,15 @@ class _UIRenderer(BaseRenderer):
 
     def render(self, swagger, accepted_media_type=None, renderer_context=None):
         if not isinstance(swagger, Swagger):  # pragma: no cover
-            # if `swagger` is not a ``Swagger`` object, it means we somehow got a non-success ``Response``
-            # in that case, it's probably better to let the default ``TemplateHTMLRenderer`` render it
-            # see https://github.com/axnsan12/drf-yasg/issues/58
-            return TemplateHTMLRenderer().render(swagger, accepted_media_type, renderer_context)
+            try:
+                # if `swagger` is not a ``Swagger`` object, it means we somehow got a non-success ``Response``
+                # in that case, it's probably better to let the default ``TemplateHTMLRenderer`` render it
+                # see https://github.com/axnsan12/drf-yasg/issues/58
+                return TemplateHTMLRenderer().render(swagger, accepted_media_type, renderer_context)
+            except ImproperlyConfigured:
+                # Fall back to using eg '404 Not Found'
```

Comment:
Stolen from the error handling code path https://github.com/encode/django-rest-framework/blob/f0a5b958a134e8cd94e3ef3263e8fa623ac9b82f/rest_framework/renderers.py#L199-L202